File: androidApp/src/main/java/dev/emanuelepapa/kmmsurvivalguidesamples/android/inlineclasses/InlineClassesScreen.kt
package dev.emanuelepapa.kmmsurvivalguidesamples.android.inlineclasses

import androidx.compose.foundation.layout.Column
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState

@Composable
fun InlineClassesScreen(viewModel: InlineClassesScreenViewModel) {
    val state = viewModel.state.collectAsState().value
    Column {
        Button(onClick = { viewModel.onShowMyNameClicked() }) {
            Text(text = "Show my name!")
        }
        val text = state.formattedName?.let {
            "My name is: $it"
        } ?: "My name is..."
        Text(text = text)
    }
}

File: iosApp/iosApp/inlineclasses/InlineClassesScreen.swift
//
//  InlineClassesScreen.swift
//  iosApp
//
//  Created by Emanuele on 16/09/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import shared

struct InlineClassesScreenView: View {
    
    @ObservedObject var viewModel: InlineClassesScreenViewModel
    
    var body: some View {
        VStack {
            Button {
                viewModel.onShowMyNameClicked()
            } label: {
                Text("Show my name!")
            }
            Text(viewModel.state.formattedName ?? "My name is...")
            Button {
                viewModel.onShowMyNameUsingInlineClassesClicked()
            } label: {
                Text("Show my name using Inline Classes!")
            }
            Text(viewModel.state.formattedNameUsingInlineClasses ?? "My InlineClasses name is")
        }
    }
}
